packages/ingest/readers/gmail.py

```
@@ -111,3 +139,455 @@ def load_data(self, query: str = "", limit: int | None = None) -> list[Document]
+
+        Args:
+            thread: Thread object from Gmail API.
+
+        Returns:
+            Thread entity.
+        """
+        now = datetime.now(UTC)
+        messages = thread.get("messages", [])
+
+        if not messages:
+            raise ValueError(f"Thread {thread['id']} has no messages")
```

Ruff (0.14.2)

Avoid specifying long messages outside the exception class

(TRY003)